libraries/AP_RCProtocol/AP_RCProtocol_IBUS.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Code by Sidhant Goel
+ */
+/*
+  IBUS decoder based on SBUS decoder
+ */
+
+#include "AP_RCProtocol_IBUS.h"
+
+#define IBUS_FRAME_SIZE		32
+#define IBUS_INPUT_CHANNELS	14
+
+// constructor
+AP_RCProtocol_IBUS::AP_RCProtocol_IBUS(AP_RCProtocol &_frontend, bool _inverted) :
+    AP_RCProtocol_Backend(_frontend),
+    inverted(_inverted)
+{}
+
+/*
+  process a IBUS input pulse of the given width
+ */
+void AP_RCProtocol_IBUS::process_pulse(uint32_t width_s0, uint32_t width_s1)
+{
+    uint32_t w0 = width_s0;
+    uint32_t w1 = width_s1;
+    if (inverted) {
+        w0 = saved_width;
+        w1 = width_s0;
+        saved_width = width_s1;
+    }
+    uint8_t b;
+    // uint16_t values[6] = {1500, 1500, 1500, 1500, 1500, 1500};
+    //     add_input(6, values, false);
+    //     #if HAL_OS_POSIX_IO
+    //     #error 1
+    //     #endif
+    if (ss.process_pulse(w0, w1, b)) {
+        //uint16_t values[6] = {1000, 1000, 1000, 1000, 1000, 1000};
+        //add_input(6, values, false);
+        _process_byte(b);
+    }
+}
+
+// decode a full IBUS frame
+bool AP_RCProtocol_IBUS::ibus_decode(const uint8_t frame[32], uint16_t *values, uint16_t *nvalues, uint16_t max_values)
+{
+    int i;
+    /* check frame boundary markers to avoid out-of-sync cases */
+    if ((frame[0] != 0x20)) {
+        return false;
+    }
+    if ((frame[1] != 0x40)) {
+        return false;
+    }
+    uint16_t checksum = (frame[31] << 8) | frame[30];
+    uint16_t calcsum = 0xFFFF;
+    for(i = 0; i < 30; i++) calcsum -= frame[i];
+    if(checksum != calcsum) {
+        uint16_t values1[IBUS_INPUT_CHANNELS] = {1000, 1400, 1000, 1000, 1000, 1000};
+        add_input(6, values1, false);
+       return false;
+    }
+
+    *nvalues = max_values;
+    if(*nvalues > 14) *nvalues = 14;
```

Comment:
Please put this on a seperate line and use `{ }` as per the style guide.

Reply:
This is also clearer to be written as `*nvalues = MIN(max_values, 14)` rather then the awkward pointer dereference/if.

Reply:
Also use your define, rather then the magic value of 14.

Reply:
ok